docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,552 @@
+## API endpoints
+
+The CMC exposes a set of core methods for converting ICP into cycles and interacting with subnet configuration. These include:
+
+-   `notify_create_canister`: Processes an ICP payment by minting cycles and using them to create a new canister, assigning control to the specified principal and applying optional settings.
+-   `notify_top_up`: Processes an ICP payment by minting cycles and sending them to an existing canister to increase its available balance.
+-   `notify_mint_cycles`: Processes an ICP payment by minting cycles and depositing them into the caller's cycles ledger account associated with a subaccount.
+-   `create_canister`: Creates a canister using cycles directly attached to the call.
+-   `get_icp_xdr_conversion_rate`: Returns the current ICP/XDR exchange rate with certification.
+-   `get_subnet_types_to_subnets`: Lists publicly available subnets grouped by their *named types* (assigned by governance).
+-   `get_principals_authorized_to_create_canisters_to_subnets`: Indicates which principals are permitted by governance to create canisters on specific, non-default subnets.
+-   `get_default_subnets`: Returns the subnets designated by governance for general-purpose public canister creation.
+-   `total_cycles_minted`: Returns the total number of cycles ever minted by the CMC.
+-   `get_build_metadata`: Displays internal version and build information for the CMC.
+-   `http_request`: Provides canister metrics via a standard HTTP interface (useful for monitoring).
+
+
+## `notify_create_canister`
+
+Uses tokens from an ICP transfer to **mint cycles**, which are then used to **create a new canister**. The CMC first verifies that the transfer has been recorded in the ICP ledger and that it matches the expected structure. If the `subnet_selection` field is omitted, the CMC selects a suitable subnet at random from the list of subnets available to the `controller` (either default or authorized).
+
+The CMC expects the payment to follow a strict structure that encodes both the **intent** and the **recipient**:
+
+-   The **destination account** of the ICP transfer must be the CMC's account with a **subaccount derived from the intended controller's principal** (see the section on [Encoding a principal into a subaccount](#encoding-a-principal-into-a-subaccount)).
+-   The **principal encoded in the subaccount is the intended controller** of the created canister.
+-   Only that principal is authorized to call `notify_create_canister` and can set **arbitrary canister settings**, including additional controllers.
+-   **Exception**: The [NNS dapp](https://nns.ic0.app/) is permitted to call `notify_create_canister` on behalf of users and acts as a proxy in that case.
+-   The **memo field** must explicitly indicate the intent to create a canister. This can be expressed as:
+    -   A legacy 64-bit unsigned integer memo with value (in decimal):
+        ```
+        1095062083
```

Comment:
How was this derived?  It would be good to have an explanation, so it can be checked.  I believe it should work out to "CREA" but in little endian if I remember correctly.